ic-agent/src/agent/route_provider/dynamic_routing/dynamic_route_provider.rs

```diff
@@ -7,19 +7,15 @@ use std::{
 
 use arc_swap::ArcSwap;
 use candid::Principal;
+use futures_util::FutureExt;
 use reqwest::Client;
+use stop_token::StopSource;
 use thiserror::Error;
-use tokio::{
```

Comment:
@adamspofford-dfinity even though `futures_util` can run with `tokio` runtime, i'm not sure the performance will be as good as with pure `tokio`.  As `tokio` specific utilities like tasks are optimized for `tokio`, so i'm a bit concerned with the decision to remove it.

Reply:
i get the rationale to make the code work for `wasm` builds too, but don't like that standard `build` will potentially loose in performance